lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1673,6 +1674,52 @@ bool TypeChecker::typeCheckFunctionBodyUntil(FuncDecl *FD,
   BraceStmt *BS = FD->getBody();
   assert(BS && "Should have a body");
 
+  // FIXME: FIXME_NOW: DETERMINE: What is the appropriate place for this code to
+  //        live?
+  if (FD->hasSingleExpressionBody()) {
+    auto fnTypeLoc = FD->getBodyResultTypeLoc();
+    auto fnType = fnTypeLoc.getType();
+    auto returnStmt = cast<ReturnStmt>(BS->getElement(0).get<Stmt *>());
+    auto E = returnStmt->getResult();
+
+    // FIXME: FIXME_NOW: DETERMINE: Does this actually need to be here?  It's 
+    //        being done for closures, but there may be a reason that applies
+    //        there but not here.
+    //
+    //           if (E != FD->getSingleExpressionBody()) {
+    //             FD->setSingleExpressionBody(E);
+    //           }
+
+    if (fnTypeLoc.isNull() || fnType->isVoid()) {
```

Comment:
You've already asked for `getType()`, so you know the TypeLoc isn't null.

Reply:
Fixed so that `getType()` isn't called before checking that `fnTypeLoc.isNull()` is false.